src/main/java/ru/mts/hw/repository/impl/AnimalsRepositoryImpl.java

```diff
@@ -0,0 +1,183 @@
+package ru.mts.hw.repository.impl;
+
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Component;
+import ru.mts.hw.animal.Animal;
+import ru.mts.hw.repository.AnimalsRepository;
+import ru.mts.hw.service.CreateAnimalService;
+
+import javax.annotation.PostConstruct;
+import java.math.BigDecimal;
+import java.time.LocalDate;
+import java.time.Period;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+@Component
+@RequiredArgsConstructor
+public class AnimalsRepositoryImpl implements AnimalsRepository {
+    private Map<String, List<Animal>> animalsMap;
+    private final CreateAnimalService createAnimalService;
+
+    @PostConstruct
+    public void postConstruct() {
+        animalsMap = createAnimalService.createAnimals();
+    }
+
+
+    /**
+     * Метод для нахождения в массиве животных имён тех животных,
+     * которые родились в вискосный год
+     *
+     * @return Массив объектов класса String - имён животных
+     */
+    public Map<String, LocalDate> findLeapYearNames() {
+        validateAnimals();
+
+        Map<String, LocalDate> leapYearNames = new HashMap<>();
+
+        animalsMap.entrySet()
+                .stream()
+                .forEach(pair -> pair.getValue()
+                        .stream()
+                        .filter(animal -> animal.getBirthDate().isLeapYear())
+                        .forEach(rightAnimal -> leapYearNames.put(
+                                        pair.getKey().concat(" ".concat(rightAnimal.getName())), rightAnimal.getBirthDate()
+                                )
+                        )
+                );
```

Comment:
использование `.**stream**().forEach` тут бессмысленно, по сути. стрим ради стрима.
можно было бы использовать и стандартный форич, например так:

```suggestion
        animalsMap.forEach((key, value) -> value
                .stream()
                .filter(animal -> animal.getBirthDate().isLeapYear())
                .forEach(rightAnimal -> leapYearNames.put(
                                key.concat(" ".concat(rightAnimal.getName())), rightAnimal.getBirthDate()
                        )
                ));
```

я думаю, имелось в виду вообще отказаться от итеративных методов в угоду стрима.
но, тоже приемлимо.